libyul/backends/evm/AbstractAssembly.h

```diff
@@ -112,6 +112,9 @@ class AbstractAssembly
 	/// Appends an assignment to an immutable variable.
 	virtual void appendImmutableAssignment(std::string const& _identifier) = 0;
 
+	/// Appends 32 bytes data load from EOF data section in dataOffset pos
```

Comment:
This is inaccurate. The function reads from the aux data, not the whole data section. It would also be good to provide a bit more detail. How about this:

```suggestion
	/// Appends an operation that loads 32 bytes of data from a known offset relative to the start of the static_aux_data area of the EOF data section.
	/// Note that static_aux_data is only a part or the data section.
	/// It is preceded by the pre_deploy_data, whose size is not determined before the bytecode is assembled, and which cannot be accessed using this function.
	/// The function is meant to allow indexing into static_aux_data in a way that's independent of the size of pre_deploy_data.
```

Reply:
We should also add a note to `appendToAuxiliaryData()` below to make it clear that EOF aux data and our auxiliary data are different things.